radicle-cli/src/git.rs

```diff
@@ -175,24 +175,48 @@ pub fn is_signing_configured(repo: &Path) -> Result<bool, anyhow::Error> {
 }
 
 /// Return the list of radicle remotes for the given repository.
-pub fn remotes(repo: &git2::Repository) -> anyhow::Result<Vec<(String, NodeId)>> {
-    let mut remotes = Vec::new();
-
-    for name in repo.remotes().iter().flatten().flatten() {
-        let remote = repo.find_remote(name)?;
-        for refspec in remote.refspecs() {
-            if refspec.direction() != git2::Direction::Fetch {
-                continue;
+pub fn rad_remotes(repo: &git2::Repository) -> anyhow::Result<Vec<git2::Remote>> {
+    let remotes: Vec<_> = repo
+        .remotes()?
+        .iter()
+        .filter_map(|name| {
+            let Some(name) = name else {
+                return None
+            };
+            let Some(remote) = repo.find_remote(name).ok() else {
+                return None
+            };
+
+            let Some(url) = remote.url() else {
+                return None
+            };
+
+            if url.starts_with(radicle::storage::git::transport::local::Url::SCHEME) {
+                Some(remote)
+            } else {
+                None
             }
```

Comment:
```rust
.filter_map(|name| {
    name.and_then(|name| repo.find_remote(name).ok())
        .and_then(|remote| remote.url())
        .and_then(|url| {
            if url.starts_with(radicle::storage::git::transport::local::Url::SCHEME) {
                Some(remote)
            } else {
                None
            }
        })
})
```
... or something like that is a bit cleaner

Reply:
Make sense, I think we need the full remove object so I think this is the best that we can do?

```diff
diff --git a/radicle-cli/src/git.rs b/radicle-cli/src/git.rs
index a9198df..6f0bedd 100644
--- a/radicle-cli/src/git.rs
+++ b/radicle-cli/src/git.rs
@@ -180,22 +180,18 @@ pub fn rad_remotes(repo: &git2::Repository) -> anyhow::Result<Vec<git2::Remote>>
         .remotes()?
         .iter()
         .filter_map(|name| {
-            let Some(name) = name else {
-                return None
-            };
-            let Some(remote) = repo.find_remote(name).ok() else {
-                return None
-            };
-
-            let Some(url) = remote.url() else {
-                return None
-            };
-
-            if url.starts_with(radicle::storage::git::transport::local::Url::SCHEME) {
-                Some(remote)
-            } else {
-                None
-            }
+            name.and_then(|name| repo.find_remote(name).ok())
+                .and_then(|remote| {
+                    let Some(url) = remote.url() else {
+                        return None
+                    };
+
+                    if url.starts_with(radicle::storage::git::transport::local::Url::SCHEME) {
+                        Some(remote)
+                    } else {
+                        None
+                    }
+                })
         })
         .collect();
     Ok(remotes)
```